### PR TITLE
64 bit support for OS X, Windows

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/openrct2.c [hunk 1]
```diff
@@ -499,7 +503,7 @@ bool openrct2_setup_rct2_segment()
 {
 	// OpenRCT2 on Linux and macOS is wired to have the original Windows PE sections loaded
 	// necessary. Windows does not need to do this as OpenRCT2 runs as a DLL loaded from the Windows PE.
-#if defined(__unix__)
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
```
Comment (marijnvdwerf) [inline src/openrct2.c]: Isn't there __OSX__ or __MACOS__ for this?

Comment (janisozaur) [inline src/openrct2.c]: Not according to https://sourceforge.net/p/predef/wiki/OperatingSystems/

**Comment by AaronVanGeffen** [comment]
Nice work, it renders the title screen nicely ­— barring the tracks that lack a paint implementation. Very nice progress indeed.

For what it's worth, it crashes when clicking New Game, or when loading a scenario directly from the command line.

**Comment by janisozaur** [comment]
@AaronVanGeffen I've rebased it on top of #4245 can you check now?

**Comment by AaronVanGeffen** [comment]
@janisozaur New Game works now; loading a scenario does not.

File: src/openrct2.c (hunk 1 above)
Comment (marijnvdwerf) [inline src/openrct2.c]: `__MACOSX__` is used more widely in the codebase.

**Comment by janisozaur** [comment]
@AaronVanGeffen it should work now

**Comment by AaronVanGeffen** [comment]
@janisozaur It does! :+1:

**Comment by janisozaur** [comment]
This branch now contains changes that can produce 64 bit Windows binary. There is a small issue, where I had to manually add `-liconv` to the final linking step.

**Comment by janisozaur** [comment]
This shouldn't be WIP any more, but for the appveyor issue I can't figure out. Can you guys please check if it still works for you in 32 and 64 bit mode?

@IntelOrca appveyor thing looks like something easy I missed, can you fix that? Should be mergeable then.

File: src/openrct2.c [hunk 2]
```diff
@@ -611,7 +611,6 @@ bool openrct2_setup_rct2_segment()
 #endif // defined(__unix__)
 
 #if defined(USE_MMAP) && defined(__WINDOWS__)
-	#error
```
Comment (janisozaur) [inline src/openrct2.c]: I was testing something and it was late…

**Comment by janisozaur** [comment]
This includes #4197. @LRFLEW are you OK with merging it?

File: CMakeLists.txt [hunk 3]
```diff
@@ -63,6 +63,10 @@ add_definitions(-DOPENRCT2_COMMIT_SHA1_SHORT="${OPENRCT2_COMMIT_SHA1_SHORT}")
 
 INCLUDE(FindPkgConfig)
 
+if (APPLE)
+	set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/local/opt/openssl/lib/pkgconfig")
+endif (APPLE)
```
Comment (AaronVanGeffen) [inline CMakeLists.txt]: Hmm, OpenSSL has worked just fine for me without this change. Had a bit of trouble getting it recognised by cmake after installing it through brew, but in the end it worked as long as it was brewn universally.

**Comment by JarnoVgr** [comment]
Is this going to be added to openrct2.org download listing with a WIP label?

**Comment by AaronVanGeffen** [comment]
Given most roller coasters won't be painted in the 64-bits versions yet, I'd advise against it at this point.

**Comment by janisozaur** [comment]
I don't think so, at least not yet. I believe the people who would be interested in such experimental builds at this point are the same people who roll their own builds. I can upload them to https://transfer.sh where others can download them.

Also, I mentioned this on gitter few moments ago:
> 64 bit jobs have much faster turnaround time, due to not having to install half the system
> Once they're all fixed I may switch most jobs to 64 bit and only leave ones that create builds 32 bit

**Comment by LRFLEW** [comment]
I wish that when you want my OK on a PR, that you wait until I'm actually awake before merging it 😜.

I took a look at it, and it looks pretty good to me. I noticed a seg-fault when clicking on exit on the main menu, but otherwise it seemed fine. If you want me to update the Xcode project to support 64-bit builds, I could try. Do remember, though, that getting Xcode to work with system-installed libraries isn't easy, and I'd end up making a 64-bit build of the dependencies to get it to work.

File: CMakeLists.txt (hunk 3 above)
Comment (LRFLEW) [inline CMakeLists.txt]: @AaronVanGeffen If by "brewn universally", you meant force-linked (aka `brew link --force openssl`), then this avoids having to do that. It's a terrible idea to require a library to be force linked (there's a reason it isn't linked by default), and this more closely follows the solution suggested by brew itself.

**Pull request merged**